std/algorithm/iteration.d

```diff
@@ -902,67 +913,100 @@ template each(alias pred = "a")
             size_t i = 0;
             while (!r.empty)
             {
-                cast(void) binaryFun!BinaryArgs(i, r.front);
+                static if (!is(typeof(binaryFun!BinaryArgs(i, r.front)) ==
+                        Flag!"each"))
+                    cast(void) binaryFun!BinaryArgs(i, r.front);
```

Comment:
@MetaLang hmmm, yah but in that case it seems the cast precludes a potentially useful warning. This also ties in somehow with @schveiguy's discussion of `free`.